### DISCUSSION TITLE
Vote for more Databases

### DISCUSSION THREAD

**Discussion by kiwicopple** [supabase/supabase#6] [Feature Requests]
This issue is for tracking requests/demand for different databases. Give a comment a thumbs up if you want us to build Supabase for your favourite database.

Please refrain from adding comments unless you are adding a database which isn't already listed.

**Comment by kiwicopple** [comment]
PostgreSQL - supported!!

**Comment by kiwicopple** [comment]
MySQL

**Reply by jdgamble555** [reply]
From my understanding, mySQL should be the main database instead of postgreSQL, since it is not a distributed database.  Supabase claims to be Firestore's competitor, but postgreSQL does not scale well horizontally.

**Reply by organicnz** [reply]
Would be Supa useful as there are so many WordPress websites that would def switch to Supabase MySQL :)

**Comment by kiwicopple** [comment]
SQLite

**Reply by Fanna1119** [reply]
sqlite is awesome for tiny projects, especially static sites and blogs, small-medium sized databases.
You often don't need a postgres and I think allowing sqlite would be an awesome addition.

**Reply by rnbokade** [reply]
Absence of postgrest like project for same will be the biggest issue I guess.

**Reply by barakplasma** [reply]
Directus manages to provide graphql from sqlite

**Reply by c00kiemon5ter** [reply]
Litestream is now a thing atop sqlite
https://litestream.io/

**Reply by tunecino** [reply]
This could help with offline support (https://github.com/supabase/supabase/discussions/357), specially after SQLite adding JSON type the same way PostgreSQL is handling it. With raising tools like Tauri already [working on mobile support](https://tauri.app/blog#whats-next), SQLite & ways to sync to PostgreSQL may be huge for the future.

**Reply by CanRau** [reply]
Just wanted to mention that this might be interesting for support of [Cloudflare D1](https://blog.cloudflare.com/introducing-d1/)

**Comment by ajwerner** [comment]
CockroachDB

Cockroach is PostgreSQL wire compatible and has good and constantly increasing compatibility. Generally in places where there are incompatibilities there are easy workarounds. It should be very easy to support, in fact, it might just work. Cockroach has online upgrades, online schema changes, and is replicated for high availability and some other advantages over Postgres.

**Reply by haladynkamil** [reply]
Anything new on this? Im trying to integrate cockroachdb with supabase

**Reply by spencerbart** [reply]
This would be awesome. We are looking for something super scalable.

**Reply by KFree98** [reply]
Has Anyone tested the plug and play compatibility on crdb with supabase?

**Reply by clibequilibrium** [reply]
Sadly cockroachdb has not RLS yet...

**Reply by Rizzist** [reply]
Still Waiting for this
Only thing keeping me from jumping unto the Supabase Boat!

**Reply by vim951** [reply]
+1

**Comment by JCKodel** [comment]
MSSQL

**Reply by tonyborchert** [reply]
Its closed source, so there's no point

**Reply by huperniketes** [reply]
If Supabase hopes to be used by enterprise shops, support for existing enterprise infrastructure will be necessary.

**Comment by kiwicopple** [comment]
Thanks for all the interest here. 

Just a small update that we are 100% committed to Postgres right now. And we're deeply integrating with some of the features Postgres has to offer (particularly auth). Cockroach is interesting to us since it's wire-compatible, but we still have a lot of features to build and supporting even 2 databases will slow us down immensely. 

Keep the suggestions coming though! When we reach 1.0 we will loop back on this one

**Reply by spencerbart** [reply]
Cockroach would be awesome. Any updates on Cockroach?

**Reply by nbaua** [reply]
> Thanks for all the interest here.
> 
> Just a small update that we are 100% committed to Postgres right now. And we're deeply integrating with some of the features Postgres has to offer (particularly auth). Cockroach is interesting to us since it's wire-compatible, but we still have a lot of features to build and supporting even 2 databases will slow us down immensely.
> 
> Keep the suggestions coming though! When we reach 1.0 we will loop back on this one

In that case. it seems you guys are a bit early. I liked your enthusiasm though. Keep the ball rolling.

**Comment by aridwiprayogo** [comment]
Redis

**Reply by jdgamble555** [reply]
redisgraph if we go that way!

**Reply by ioucyf** [reply]
redis is a must in my opinion, and the entire redis stack if possible! 🙏🙏

**Comment by itsezc** [comment]
ArangoDB

**Comment by gpproton** [comment]
Oracle

**Reply by tonyborchert** [reply]
This is (as well as Microsoft SQL) closed source, so there's no point

**Reply by huperniketes** [reply]
If Supabase hopes to be used by enterprise shops, support for existing enterprise infrastructure will be necessary.

**Comment by mmpneo** [comment]
Mnesia

**Comment by juan040** [comment]
DynamoDB

**Comment by miso-belica** [comment]
https://www.yugabyte.com/ is also [compatible with PG](https://blog.yugabyte.com/postgresql-compatibility-in-yugabyte-db-2-0/) but has some [features](https://www.yugabyte.com/yugabytedb-vs-cockroachdb/) [similar to](https://www.cockroachlabs.com/blog/unpacking-competitive-benchmarks/) [CockroachDB](https://blog.yugabyte.com/yugabytedb-vs-cockroachdb-bringing-truth-to-performance-benchmark-claims-part-1/).

**Reply by ronasuti** [reply]
This is a really great one, as YSQL has even greater compatibility than CDB. For example, Hasura works OOTB with YSQL, but currently doesn’t support CDB, mainly as Hasura needs features like triggers. If CDB is not working for those reasons, YSQL probably would. It’s only problem is that it does not support gist indexes at all which could be a problem for spatio-temporal tables (GIS, reservation systems, etc.)

**Reply by FranckPachot** [reply]
YugabyteDB is the most PostgreSQL compatible within the distributed SQL databases, as it reuses postgres code, but there's still a few things in progress that are used by supabase (PG14 features, GIST, some extensions). @kiwicopple I don't know how those features are mandatory or can be worked around but could be interesting to list them

**Reply by kneeraj** [reply]
We ( Yugabyte ) are currently exploring YugabyteDB with Supabase ( [YugabyteDB Github Issue](https://github.com/yugabyte/yugabyte-db/issues/12714) ). There have been few requests for this from the community on our side as well.

**Reply by wawilation** [reply]
My vote is with YugabyteDB as well.

**Reply by dcaoile** [reply]
Raising my hand and standard for YugabyeDB! 🚀

**Comment by benmccann** [comment]
AWS RDS & Aurora

**Reply by tonyborchert** [reply]
How is that even supposed to work with an open source Projekt like Supabase?

**Reply by okadts** [reply]
Same question here. I also ask the AWS support and they are not sure that it will be compatible.
Even though I send them this link https://supabase.com/blog/2021/07/26/epsilon3-self-hosting where it said that use AWS RDS.
The AWS support team encourage us ask back if the third party (Supabase) support AWS RDS or just use the EC2.
Any though on this @kiwicopple , should we use the EC2 (docker method).
Thanks

**Comment by stephanelpaul** [comment]
> Mnesia

oufff erlang stuff

**Comment by duncanhealy** [comment]
MongoDB

**Reply by arthurmougin** [reply]
MongoDB Atlas Search is a league ahead of Postgres FTS, yet lack the developer experience of Supabase, self-documenting API. Connecting both would allow us to take the best of both world.

**Comment by stevefan1999-personal** [comment]
Cosmos DB

**Comment by likern** [comment]
@kiwicopple Can you consider YugaByte distributed SQL database.

They use PostgreSQL SQL engine and claim compatibility with PostgreSQL (it's drivers can be used)

**Reply by miso-belica** [reply]
If you mean https://www.yugabyte.com/ it is mentioned at https://github.com/supabase/supabase/discussions/6#discussioncomment-80183 already :)

**Comment by meatflavourdev** [comment]
[GunDB](https://gun.eco/)-- hey, let's throw a little bit of everythin' in the hat.  🎩👈🔫

**Reply by fangmarks** [reply]
I always wanted to try gun but never had a use for it really, so this'd be really neat 👀

**Reply by meatflavourdev** [reply]
Let's be frank: the docs suck.

**Reply by marcusmmmz** [reply]
This doesn't make sense though, GunDB is decentralized, how would you even integrate it into Supabase, which is centralized?

**Comment by eikooc** [comment]
Dgraph

**Reply by likern** [reply]
I think Dgraph on it's own is a competitor. It's not only database, but Ggraph provides a service like SupaBase and FireBase.

**Reply by eikooc** [reply]
@likern dgraph the company provides a hosted version of the open-source database dgraph, but Supabase provides a lot more than that on top. Supporting a graph database besides normal SQL databases would be a really cool addition.

**Reply by fungilation** [reply]
I'm actually thinking about both Dgraph as a service, and something like firebase/supabase. What supabase (and firebase) is not is it's not in the Graphql world with all the great tooling in that ecosystem, for DX as well as client optimizations (no under or overfetching).

What if we create another OSS lib that creates higher level abstraction on top of Dgraph, that provides a turnkey graphql as a service? Gql schema as input, full CRUD gql api + auth (and what's standard in 90% of web apps).

Thoughts? Interests on collaborating?

**Comment by yaroslav-ilin** [comment]
Clickhouse

**Comment by nbaua** [comment]
RethinkDB

**Comment by nstrelow** [comment]
Apache Cassandra 👁️

Marketing speech: "Apache Cassandra is an open source NoSQL distributed database trusted by thousands of companies for scalability and high availability without compromising performance. Linear scalability and proven fault-tolerance on commodity hardware or cloud infrastructure make it the perfect platform for mission-critical data."

**Reply by tonyborchert** [reply]
This is the only good one I've seen so far, stuff like MySQL is pointless as there is so little difference between MySQL and PostgreSQL. Stuff like GunDB or Redis are also pointless, they both don't really make that much sense. But Cassandra on the Hand is pretty similar to SQL but it definitely has its own uses

**Comment by jdgamble555** [comment]
neo4j

**Reply by eikooc** [reply]
You mean **Neo4j** right?

**Comment by donPuerto** [comment]
Redis

**Reply by baderdean** [reply]
Especially a REST and  Websocket client something like PostgREST + Realtime

**Comment by phpb-com** [comment]
ScyllaDB

**Comment by ghost** [comment]
TimeScale

**Comment by ghost** [comment]
QuestDB

**Reply by o624366705** [reply]
Mysql

**Reply by killshot13** [reply]
> Mysql

https://github.com/supabase/supabase/discussions/6#discussioncomment-80173

**Comment by grind-t** [comment]
EdgeDB looks promising. It's an open-source object-relational database built on top of PostgreSQL with native GraphQL support.

**Reply by prescience-data** [reply]
Just saw this too - this would be super cool to get the EdgeQL magic along with Supabase's Auth.
Reference: https://www.edgedb.com/docs

**Reply by ild0tt0re** [reply]
Wow! In one word "the future".

**Reply by devi-prsd** [reply]
They got their own auth now and cloud, I'm hoping they speedup as fast as postgres.

**Comment by noriakisana** [comment]
InfluxDB - for timeseries

**Comment by de1mat** [comment]
CouchDB

**Comment by vphaneendrayb** [comment]
Yugabyte - for Distributed SQL

**Comment by ngaut** [comment]
TiDB - open-source, cloud-native, HTAP database for elastic scale and real-time analytics

**Comment by c00kiemon5ter** [comment]
A temporal database.

There is a different world of features and niceties around temporal databases (like `datomic`, `xtdb` or `datahike`) which I find super interesting to be able to work with within an application. The nice thing about such databases (aside from database-branching, as-of "time-travel", etc) is that you can use the same data source to do _analytics_. This would bring supabase closer to a feature-full firebase alternative.

AFAICT, datomic is propriatery, xtdb is a big complex system but there are more smaller ones coming up. I cannot recommend one, but there is an interesting thread on the different choices and features over at: https://github.com/athensresearch/athens/issues/9

It seems [biff](https://biffweb.com/) is trying to find its position in that space, but I feel supabase is more focused with better choices and tooling underneath, but most importantly an active community.

Now, to be pragmatic, I don't expect supabase to suddenly switch over to these systems, but I do think there are lessons that can be learnt and features that we could try to think how to get them through supabase and postgres, and I think that would give a lot of leverage to supabase.

**Reply by c00kiemon5ter** [reply]
Another clojure based open-souce choice is Fluree:
https://flur.ee/

**Reply by c00kiemon5ter** [reply]
TerminusDB seems to be a non-clojure open-source contender in this space:
https://github.com/terminusdb/terminusdb

**Reply by davidhmays** [reply]
Could you explain what you mean by being able to use the same datasource to do _analytics_? Can't you do analytics from any datastore, what makes the temporal databases better at this? I'm trying to decide between diving into XTDB, or learning EdgeDB, and part of that is trying to decide if I really need XTDB's advertised bitemporality, or if EdgeDB's time datatypes are all I'd really need. (And do either of these or Datomic support geospatial data?) Also trying to decide if I really need XTDB's schemaless, or am fine with EdgeDB's excellent migrations. Both seem great, but diving into one or the other is such a different paradigm instead of actually learning anything I worry about deciding on one haha. Liked the marketing behind Surreal, but the documentation isn't ready and the EdgeDB and XTDB docs are pretty good.

**Comment by c00kiemon5ter** [comment]
Alternative to Materialize?
https://materialize.com/

**Comment by BhaskaranR** [comment]
has anyone tried AlloyDB? looks to be fully compatible with postgres.

**Reply by weburnit** [reply]
I tried, it works a half. the service_role/anon is possible to query the table on postgrest.

**Reply by weburnit** [reply]
can not use logflare! because it doesn't allow to have wal_sender

**Comment by Manas-Nagelia** [comment]
SurrealDB.

**Reply by axscell** [reply]
That would be awesome if implemented!

**Comment by deezed420** [comment]
SurrealDB. That would be a nice addon to the SurrealDB app too.

**Comment by xuman2019** [comment]
[DB3 Network](https://github.com/dbpunk-labs/db3) - Decentralized Firebase Firestore Alternative.

**Comment by Shalashtein** [comment]
Fluree 

A database for collaborative clean extensible composable data built for data centric systems where agents are the largest consumers and producers of data.

JSON-LD document based
Knowledge graph triple-store
Programmable data (policies and contracts...)
Cryptographically proven identities

**Comment by arghaffari** [comment]
Neo4j
